### PR TITLE
refactor: replace REST token registry with DB-based assets-ext reads

### PR DESCRIPTION
**Pull request by matiwinnetou**
## Summary
Adds direct database access to token metadata tables (CIP-26 / CIP-68) populated by the `yaci-store-assets-ext` extension. The API module reads token metadata — name, ticker, decimals, description, logo — from the shared database via its own JPA entities and a batched query path, so `/account/balance`, `/block/transaction`, and `/search/transactions` can enrich native-token balances and operations with metadata without any out-of-process call.

### yaci-indexer (write side)
- `yaci-store-assets-ext` + `yaci-store-assets-ext-spring-boot-starter` dependencies
- `store.assets.ext.enabled=true` + `store.assets.ext.cip26.enabled=true` in `application.properties`
- CIP-68 enabled by default (inner class default); CIP-113 left disabled for now

### api module (read side)
Follows the existing project convention: own JPA entities map to yaci-store tables, no direct dependency on yaci-store jars (see "Why JPA entities are duplicated" below).

**Read model** under `api/common/model/entity/`:
- `TokenMetadataEntity` → `ft_offchain_metadata` (CIP-26)
- `TokenLogoEntity` → `ft_offchain_logo` (CIP-26)
- `MetadataReferenceNftEntity` + `MetadataReferenceNftId` → `metadata_reference_nft` (CIP-68)

**Repositories** under `api/common/model/repository/`:
- `TokenMetadataRepository`, `TokenLogoRepository`
- `MetadataReferenceNftRepository` + `MetadataReferenceNftRepositoryCustom` / `…CustomImpl` — custom fragment with a native SQL window-function query that returns the latest-slot row per `(policy_id, asset_name)` pair in a single round-trip, regardless of batch size

**`TokenQueryService`**:
- `@Transactional(readOnly = true)`, owns all CIP-26/CIP-68 merge logic and priority (CIP-68 > CIP-26)
- Handles CIP-68 prefix conversion transparently (`0014df10` → `000643b0`)
- Abstracts CIP standard details — consumers get a unified `TokenRegistryCurrencyData`
- Accepts `Collection<AssetFingerprint>` directly (no subject/policyId string shuffling)
- Uses `StructuredTaskScope.ShutdownOnFailure` to fork the CIP-26 metadata and logo batch queries in parallel — with a documented caveat about `@Transactional` context not propagating through virtual threads (accepted because CIP-26 rows are written only during the periodic offchain sync; read-skew risk is negligible for token metadata)
- CIP-68 logo format detection: `http(s)://`, `ipfs://`, `ar://` → `URL`; everything else (including raw base64 and `data:` URIs) → `BASE64`
- Logo fetch gated by `TOKEN_REGISTRY_LOGO_FETCH` toggle

**`TokenRegistryServiceImpl`** — thin delegate:
- Asset extraction from Rosetta domain objects (`BlockTx`, `AddressBalance`, `Utxo`, `Operation`, etc.)
- Forwards metadata lookup to `TokenQueryService.queryMetadataBatch()` with `Set<AssetFingerprint>` directly

**Config:** `TOKEN_REGISTRY_LOGO_FETCH` toggle (default `false`) — when `false`, logo columns are skipped entirely from the query path so logo blobs never leave the DB.

## Why JPA entities are duplicated (temporary)

This PR introduces **three new JPA entities + repositories** in the `api` module that map to the exact same tables `yaci-store-assets-ext` owns upstream. This looks wasteful — yaci-store already publishes `Cip26StorageReader` and `Cip68StorageReader` interfaces that do the same lookups — and it is.

The duplication is **forced** by the current repo layout: `api` and `yaci-indexer` are two separate Spring Boot modules, and the `api` module has never pulled in `yaci-store-*` jars. Every yaci-store-backed table used by a Rosetta endpoint (`address_utxo`, `block`, `transaction`, `pool_registration`, etc.) has a duplicate `*Entity` + `*Repository` downstream — this PR just adds three more to that pile for the CIP-26/CIP-68 case.

The **real fix** is to merge `api` and `yaci-indexer` into a single `rosetta-app` Spring Boot module and run yaci-store in read-only mode on the API replicas, so the api side can consume yaci-store's storage reader interfaces directly without any entity/repository duplication. Tracked in **#731**.

Until #731 ships, this PR takes the pragmatic route:
- **Own JPA entities + repositories** — matches the project convention used everywhere else in the api module
- **Own custom fragment** (`MetadataReferenceNftRepositoryCustom` + `Impl`) for the CIP-68 batch window-function query, even though the same query exists upstream in `yaci-store-assets-ext`

When #731 lands, this entire downstream entity/repository/custom-fragment layer for CIP-26/CIP-68 can be deleted in favor of `cip26StorageReader.findBySubjects(...)` and `cip68StorageReader.findBySubjects(...)` — the upstream APIs the rosetta-app will consume directly.

## Tests (~62)

| Component | Tests | Coverage |
|---|---|---|
| `TokenQueryService` | 28 | CIP-26/CIP-68 merge, prefix conversion edge cases, logo format detection (http/https/ipfs/ar/base64/data-URI), logo toggle, batch queries, empty-batch short-circuit |
| `TokenRegistryServiceImpl` | 14 | Delegation to query service, asset extraction from all Rosetta types, fetch-helper edge cases |
| `TokenRegistryMapper` | 14 | Field mapping, BASE64 vs URL logo formats, null handling, end-to-end logo pipeline |
| `DataMapper` | 7 | Rosetta `Amount` building with metadata, logo passthrough, ADA defaults, value negation |

Full api unit-test run: `761 tests, 0 failures, 21 errors` — the 21 errors are pre-existing `@SpringBootTest` context-load failures that fail identically on `main` (JDK-version/devkit-DB issue, unrelated to this PR).

## Temporary: pre-release SNAPSHOT
`yaci-store-assets-ext` is not yet released. The yaci-indexer pom references `2.1.0-pre4-d358232-SNAPSHOT`, which must be installed to the local Maven repository via `./gradlew publishToMavenLocal` from the `feature/asset-store-ext` branch of bloxbean/yaci-store. **Revert this version bump once a proper release is cut** — at which point the downstream `MetadataReferenceNftRepositoryCustom` fragment can also be removed in favor of upstream's `Cip68StorageReader.findBySubjects(...)`, and ultimately the whole downstream entity layer disappears with #731.

## Test plan
- [x] Run yaci-indexer on preprod with `store.assets.ext.enabled=true` — verify `ft_offchain_metadata`, `ft_offchain_logo`, `metadata_reference_nft` populate
- [x] Run API and verify `/account/balance` returns token metadata (`name`, `ticker`, `decimals`, etc.) from DB
- [x] Test `TOKEN_REGISTRY_LOGO_FETCH=true` vs `false` — confirm logos only appear when enabled
- [x] Verify `/block` and `/search/transactions` also carry token metadata
- [x] `mvn compile -pl api,yaci-indexer` passes
- [x] New unit tests pass (62/62)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docker-compose-api.yaml [hunk 1]
```diff
@@ -4,6 +4,8 @@ services:
     build:
       context: ./
       dockerfile: ./api/Dockerfile
+      additional_contexts:
```
Comment (matiwinnetou) [inline docker-compose-api.yaml:7]: to remove after yaci-store assets-ext module release.

File: docker-compose-indexer.yaml [hunk 2]
```diff
@@ -4,6 +4,8 @@ services:
     build:
       context: ./
       dockerfile: ./yaci-indexer/Dockerfile
+      additional_contexts:
+        host-m2: ${HOST_M2:-${HOME}/.m2/repository}
```
Comment (matiwinnetou) [inline docker-compose-indexer.yaml:8]: to remove after yaci-store assets-ext module release.

File: yaci-indexer/Dockerfile [hunk 3]
```diff
@@ -21,7 +33,16 @@ COPY ./yaci-indexer /build/yaci-indexer
 COPY ./test-data-generator /build/test-data-generator
 COPY ./.git .git
 
-RUN --mount=type=cache,target=/root/.m2 mvn -U clean package -DskipTests
+# Seed the Maven cache with artifacts from the optional host-m2 context
+# (e.g. locally-built yaci-store snapshots), then build with -nsu so Maven
+# does not try to re-resolve those snapshots from remote repos.
+RUN --mount=type=cache,target=/root/.m2 \
```
Comment (matiwinnetou) [inline yaci-indexer/Dockerfile:39]: to remove after yaci-store release

**Comment by matiwinnetou** [comment]
**Blocked:**
- Yaci-Store release that includes Assets-Ext module
- Main-Net CIP-113 release

File: yaci-indexer/Dockerfile [hunk 4]
```diff
@@ -1,18 +1,30 @@
...
+#   docker buildx build --build-context host-m2=$HOME/.m2/repository ...
+FROM scratch AS host-m2
+
 FROM ubuntu:24.04 AS build-common
 WORKDIR /build
 
+# Harden apt against transient Ubuntu-mirror hiccups (mirror sync in progress,
+# flaky DNS, stale CDN edges). Retry a few times, bump the read timeout, and
+# disable HTTP caching so a stale index doesn't get re-served.
+RUN printf 'Acquire::Retries "5";\nAcquire::http::Timeout "30";\nAcquire::https::Timeout "30";\nAcquire::http::No-Cache "true";\n' \
```
Comment (matiwinnetou) [inline yaci-indexer/Dockerfile:13]: We should remove this, there is a separate PR for this, unmerged.

**Comment by Kartiiyer12** [comment]
The deleted registry still shows the data which it should not. This is an issue in this implementation as well in the the token registry api - 
deleted PR - https://github.com/cardano-foundation/cardano-token-registry/pull/8014

Example
```
curl --location 'localhost:8083/account/balance' \
--header 'Content-Type: application/json' \
--data '{
    "network_identifier": {
      "blockchain": "cardano",
      "network": "mainnet"
    },
    "account_identifier": {
      "address": "addr1q9kxzjjarcr24hnezhye78n8jq8398pakunqt0p4d6x6u05w5y6maqdyqystgfgmdgn6z33r3z8uggrnh5ln89ytmpkqcc9s7v"
    }
  }'
```

```
{
    "block_identifier": {
        "index": 13820569,
        "hash": "93fa464224d682363dcb6249e4ff48c81847f3f563d45dc15e71d877f80a9e57"
    },
    "balances": [
        {
            "value": "61245683",
            "currency": {
                "symbol": "ADA",
                "decimals": 6
            }
        },
        {
            "value": "237879600",
            "currency": {
                "symbol": "6672414441",
                "decimals": 0,
                "metadata": {
                    "policyId": "2d8cee2c3c82664023da28372e46995daab7111070abf5fc9aee72ea",
                    "subject": "2d8cee2c3c82664023da28372e46995daab7111070abf5fc9aee72ea6672414441",
                    "name": "ADAFR Pool Reward ADA",
                    "description": "Claim your additional ADA rewards at https://adafrpool.com.",
                    "ticker": "frADA",
                    "url": "https://adafrpool.com"
                }
            }
        }
    ]
}
```

Token registry also has it - 
https://tokens.cardano.org/api/v2/subjects/2d8cee2c3c82664023da28372e46995daab7111070abf5fc9aee72ea6672414441
